src/components/BlogForm/blogForm.test.jsx

```diff
@@ -0,0 +1,120 @@
+import { afterEach, describe, expect, it, vi } from "vitest";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Provider } from "react-redux";
+import { configureStore } from "@reduxjs/toolkit";
+import { BrowserRouter } from "react-router-dom";
+import authSlice from "features/auth/authSlice";
+import BlogForm from ".";
+
+describe("BlogForm component", () => {
+  const mockSetIsAddBlogFormOpen = vi.fn();
+  const mockOnSubmit = vi.fn();
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  const initialBlogDetails = {
+    title: "Test Blog",
+    tags: ["tag1", "tag2"],
+    bannerImage: "test-image.jpg",
+    body: "Lorem ipsum dolor sit amet",
+  };
+
+  const mockStore = (initialState) =>
+    configureStore({
+      reducer: {
+        auth: authSlice,
+      },
+      preloadedState: initialState,
+    });
+
+  const initialState = {};
+
+  const store = mockStore(initialState);
+
+  const user = userEvent.setup();
+
+  it("renders form with initial values", () => {
+    render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <BlogForm
+            setIsAddBlogFormOpen={mockSetIsAddBlogFormOpen}
+            blogDetails={initialBlogDetails}
+            onSubmit={mockOnSubmit}
+          />
+        </BrowserRouter>
+      </Provider>
+    );
+
+    const titleElement = screen.getByLabelText("Title");
+    expect(titleElement).toHaveValue("Test Blog");
```

Comment:
It's a good practice to check the `prop value` or declared `variable` inside the test instead of `magic string`

```suggestion
    expect(titleElement).toHaveValue(initialBlogDetails.title);
```

Reply:
Resolved